Allow users to hide the heatmap

pablobm opened the pull request:
Closes https://github.com/openstreetmap/openstreetmap-website/issues/5804, https://github.com/openstreetmap/openstreetmap-website/pull/5853.

This is a replacement for https://github.com/openstreetmap/openstreetmap-website/pull/5853. That area of the code has changed enough that I think the original PR cannot be salvaged.

This PR adds a preference that allows users to hide the heatmap from their profiles, if they so choose. There was a question of whether this was necessary, privacy implications, LWG discussion, etc. As communicated to me, this is the decision:
- OSM have a legitimate interest to show the heatmap, so it's ok that users get it by default.
- The heatmap is not strictly necessary in order to use OSM, so users should have the ability to hide it.

Notes:
- The setting can be accessed via a big button under the heatmap:

<img width="1171" height="286" alt="The heatmap on a user's profile page. A long, horizontal button underneath, spanning the whole width of the heatmap above, reads 'Edit Heatmap'" src="https://github.com/user-attachments/assets/2b2698c6-b3b5-4c5a-a837-624a5e1b5e8d" />

- When a user chooses to hide their heatmap, absolutely nobody can see it. No moderators, no admins, not even the user themselves. This is to align with the goal of making it obvious to users what is showing and what is not on their profiles.

tomhughes commented:
> > For admins - why do they need to see it? Should this be visible to moderators too? I would think moderators are more focussed on mapping behaviour than admins, but I don't know if either group use the heatmap as part of their tasks.
> 
> I vaguely recall comments that the heatmap would be useful to DWG members, but I can’t find them anymore, or maybe it was just speculation. In any case, if the administrators don’t see a need to always see the heatmap, then let’s hide it for them too. We can always revisit if the need arises.

It's probably more useful for moderators that administrators but if we do enable it for moderators then we might as well include administrators as well.

pablobm commented on config/locales/en.yml:
```diff
@@ -2214,6 +2215,15 @@ en:
         failure: Couldn't update profile location.
+    heatmaps:
+      show:
+        title: Edit Heatmap
```
It's complicated. It leads to a whole "Heatmap settings" page where yes, the only option is to toggle it. However if it said "Toggle Heatmap Visibility" I think it would be misleading, giving the impression that a click of the button is all you need.

1ec5 replied:
We do have enough room to say “Edit Heatmap Settings”. Anyhow, I’m of half a mind to suggest replacing all these edit buttons with more discreet ✏️ buttons, like you often see on templates all over the wiki. But that should be a separate discussion. Don’t want to distract from this more important feature.

pablobm commented:
Moved the setting to its own DB column `users.public_heatmap` and renamed the button as "Configure Heatmap" (plus renames to section heading and update button).

tomhughes commented on app/controllers/users/heatmaps_controller.rb:
```diff
@@ -13,7 +13,7 @@ class HeatmapsController < ApplicationController
       @user = User.find_by(:display_name => params[:user_display_name])
 
-      if @user && (@user.visible? || current_user&.administrator?)
+      if @user&.public_heatmap? && (@user.visible? || current_user&.administrator?)
```
In that case I think this is good to merge, thanks.
